storage/innobase/handler/ha_innodb.cc

```diff
@@ -2068,8 +2068,14 @@ static inline dberr_t innobase_srv_conc_enter_innodb(row_prebuilt_t *prebuilt) {
   // innodb_thread_concurreny limit how many thread can work in innodb
   // at any given time. This limit is not applicable to wsrep-applier
   // threads given they are high priority threads.
-  if (wsrep_on(prebuilt->trx->mysql_thd) &&
-      wsrep_thd_is_BF(prebuilt->trx->mysql_thd, false))
+  // Actions executed from the sst_donor thread should not be blocked as well.
+  // sst_donor thread runs with Galera LocalMonitor acquired.
+  // If we have user threads entered innodb and about to commit, they will try
+  // to acquire LocalMonitor as well, but will block. So without letting
+  // sst_donor thread to move on here, we would end up in deadlock.
+  THD *thd = prebuilt->trx->mysql_thd;
+  if (thd && WSREP_ON &&
+      (wsrep_thd_is_applying(thd)|| thd->wsrep_applier || thd->wsrep_sst_donor))
```

Comment:
```suggestion
      (wsrep_thd_is_applying(thd) || thd->wsrep_applier || thd->wsrep_sst_donor))
```

Reply:
The earlier code made a call to `wsrep_thd_is_BF()` which used the following condition
```
114     status = (wsrep_thd_is_applying(thd) || wsrep_thd_is_toi(thd) ||                                
115     |   |   | wsrep_thd_is_in_rsu(thd)) || wsrep_thd_is_in_nbo(thd);     
```

However, in the current logic we only have 
```
(wsrep_thd_is_applying(thd)|| thd->wsrep_applier || thd->wsrep_sst_donor)
```
that means we no longer check for TOI, RSU and NBO.

Reply:
Good catch. Fixed.